gobblin-runtime/src/main/java/gobblin/runtime/scheduler/ImmediateJobSpecScheduler.java

```diff
@@ -11,101 +11,49 @@
   private final ThreadFactory _jobRunnablesThreadFactory;
-  private final JobSpecSchedulerListeners _callbacksDispatcher;
 
   public ImmediateJobSpecScheduler(Optional<Logger> log) {
-    _log = log.isPresent() ? log.get() : LoggerFactory.getLogger(getClass());
+    super(log);
     _jobRunnablesThreadFactory = (new ThreadFactoryBuilder())
         .setDaemon(false)
-        .setNameFormat(_log.getName() + "-%d")
-        .setUncaughtExceptionHandler(new LoggingUncaughtExceptionHandler(Optional.of(_log)))
+        .setNameFormat(getLog().getName() + "-thread-%d")
+        .setUncaughtExceptionHandler(new LoggingUncaughtExceptionHandler(Optional.of(getLog())))
         .build();
-    _callbacksDispatcher = new JobSpecSchedulerListeners(_log);
-  }
-
-  /** {@inheritDoc} */
-  @Override public JobSpecSchedule scheduleJob(JobSpec jobSpec, Runnable jobRunnable) {
-    return scheduleOnce(jobSpec, jobRunnable);
   }
 
   @Override
-  public JobSpecSchedule scheduleOnce(JobSpec jobSpec, Runnable jobRunnable) {
+  protected JobSpecSchedule doScheduleJob(JobSpec jobSpec, Runnable jobRunnable) {
     Thread runThread = _jobRunnablesThreadFactory.newThread(jobRunnable);
-    _log.info("Starting JobSpec " + jobSpec + " in thread " + runThread.getName());
+    getLog().info("Starting JobSpec " + jobSpec + " in thread " + runThread.getName());
     JobSpecSchedule schedule =
         DefaultJobSpecScheduleImpl.createImmediateSchedule(jobSpec, jobRunnable);
-    _callbacksDispatcher.onJobScheduled(schedule);
     runThread.start();
     return schedule;
   }
 
-  /** {@inheritDoc} */
-  @Override public void unscheduleJob(URI jobSpecURI) {
-    // No-op since no future schedules are maintained
-  }
-
-  /** {@inheritDoc} */
-  @Override public Map<URI, JobSpecSchedule> getSchedules() {
-    // No future schedules
-    return Collections.emptyMap();
-  }
-
-  @Override public void registerJobSpecSchedulerListener(JobSpecSchedulerListener listener) {
-    _callbacksDispatcher.registerJobSpecSchedulerListener(listener);
-  }
-
-  @Override
-  public void unregisterJobSpecSchedulerListener(JobSpecSchedulerListener listener) {
-    _callbacksDispatcher.unregisterJobSpecSchedulerListener(listener);
-  }
-
-  @Override
-  public List<JobSpecSchedulerListener> getJobSpecSchedulerListeners() {
-    return _callbacksDispatcher.getJobSpecSchedulerListeners();
-  }
-
-  @AllArgsConstructor
-  protected class TriggerRunnable implements Runnable {
-    private final JobSpecSchedule jobSchedule;
-
-    @Override public void run() {
-      _callbacksDispatcher.onJobTriggered(jobSchedule);
-      this.jobSchedule.getJobRunnable().run();
-    }
-  }
-
   @Override
-  public void registerWeakJobSpecSchedulerListener(JobSpecSchedulerListener listener) {
-    _callbacksDispatcher.registerWeakJobSpecSchedulerListener(listener);
+  protected void doUnschedule(JobSpecSchedule existingSchedule) {
+    // nothing to do
```

Comment:
The removal of the schedule happens inAbstractJobSpecScheduler . 

Yes, it is a immediately once scheduler.